src/plugins/intel_gpu/include/intel_gpu/runtime/format.hpp

```diff
@@ -0,0 +1,335 @@
+#include <algorithm>
+#include <vector>
+#include <string>
+#include <utility>
+#include <stdexcept>
+
+namespace cldnn {
+/// @addtogroup cpp_api C++ API
+/// @{
+
+/// @addtogroup cpp_memory Memory description and management
+/// @{
+
+/// @brief Format information helper class.
+struct format_traits {
+    /// @brief String representation of a format.
+    std::string str;
+    /// @brief Number of batch dimensions in a format.
+    size_t batch_num;
+    /// @brief Number of feature map/channel dimensions in a format.
+    size_t feature_num;
+    /// @brief Number of spatial (x,y) dimensions in a format.
+    size_t spatial_num;
+    /// @brief Number of groups in a format.
+    size_t group_num;
+    /// @brief Dimensions order. Default {0, 1, 2, ... rank }
+    std::vector<size_t> _order;
+    /// @brief Dimensions changing order from rare to often.
+    std::string order;
+    /// @brief Dimensions order for internal storage.
+    std::string internal_order;
+    /// @brief Block sizes as a vector of pairs of dimension number and block size ordered from rare to often.
+    std::vector<std::pair<size_t, int>> block_sizes;
+    /// @brief Characters representing batch dimensions in an order.
+    static const char* batch_chars() { return "bno"; }
+    /// @brief Characters representing feature map/channel dimensions in an order.
+    static const char* feature_chars() { return "fic"; }
+    /// @brief Characters representing spatial dimensions in an order.
+    static const char* spatial_chars() { return "xyzhsw"; }
+    /// @brief Characters representing group dimensions in an order.
+    static const char* group_chars() { return "g"; }
+    /// @brief Checks if @p c represents batch dimension.
+    static bool is_batch_char(char c) { return std::string(batch_chars()).find_first_of(c) != std::string::npos; }
+    /// @brief Checks if @p c represents feature map/channel dimension.
+    static bool is_feature_char(char c) { return std::string(feature_chars()).find_first_of(c) != std::string::npos; }
+    /// @brief Checks if @p c represents spatial dimension.
+    static bool is_spatial_char(char c) { return std::string(spatial_chars()).find_first_of(c) != std::string::npos; }
+    /// @brief Checks if @p c represents group dimensions.
+    static bool is_group_char(char c) { return std::string(group_chars()).find_first_of(c) != std::string::npos; }
+};
+
+/// @brief Represents memory formats (orders).
+/// @n In CNN most of data is described as 4 dimensional blocks. In Intel(R) clDNN library we describe memory with 4 letters
+/// - b - number of blocks in batch. For weights formats: output features - conv, neurons - inner product
+/// - f - number of feature maps, features or channels. For weights formats: input features - conv, inputs, inner product
+/// - x - spatial, width
+/// - y - spatial, height
+/// /n
+/// For explanation how each format type is implemented in memory we will use naming shown bellow (b=2,f=3,y=3,x=3):
```

Comment:
BTW, what does this mean: `b=2,f=3,y=3,x=3`?

Reply:
I don't have any idea. Removed this part